### PR TITLE
Add support for add_note()

### PR DESCRIPTION
**Pull request by arjenzorgdoc**
See issue #3050

This adds support for `__notes__` and adds them to the exception value.

### CODE REVIEW

File: sentry_sdk/utils.py [hunk 1]
```diff
@@ -655,11 +655,15 @@ def get_errno(exc_value):
...
     # type: (Optional[BaseException]) -> str
-    return (
+    value = (
         getattr(exc_value, "message", "")
         or getattr(exc_value, "detail", "")
         or safe_str(exc_value)
     )
+    notes = getattr(exc_value, "__notes__", [])
+    if notes:
+        value = "\n".join([value] + [safe_str(note) for note in notes])
```
Comment (szokeasaurusrex) [inline sentry_sdk/utils.py:665]: I am unsure whether it makes sense to add the note to the exception message; perhaps setting this as [extra data](https://docs.sentry.io/concepts/key-terms/enrich-data/) would be better

**Comment by arjenzorgdoc** [comment]
The use case for the notes is to add text to the exception massage. 

I think `notes_new()` is the new way to do `notes_old()`:

```python
def failure():
    raise ValueError("Nope.")


MESSAGE = """\
Nope. It's wrong!
See https://example.com/why-nope.html for more info
"""


def notes_old():
    try:
        failure()
    except Exception as e:
        raise RuntimeError(f"{e}\n{MESSAGE}\n") from e


def notes_new():
    try:
        failure()
    except Exception as e:
        e.add_note(MESSAGE)
        raise
```

`notes_old()` renders like this:

image

I tried to have `notes_new()` render the same.

BTW python cli renders the notes after the Exception message as well:

```
  File ".../sentry_notes.py", line 7, in failure
    raise ValueError("Nope.")
ValueError: Nope.
Nope. It's wrong!
See https://example.com/why-nope.html for more info
```

File: sentry_sdk/utils.py (hunk 1 above)
Comment (arjenzorgdoc) [reply sentry_sdk/utils.py:723]: I responded in the pull request conversation.

I think the notes add context that developers expect to be shown directly with the error message. That is what CPython does in the console.

Comment (szokeasaurusrex) [reply sentry_sdk/utils.py:723]: I don't think we need this `safe_str` stuff here, notes should anyways just be strings. This is guaranteed when using `add_note`. Although you correctly point out in your test that it is possible to add non-string values to the `__notes__` list manually, this is likely a misuse of `__notes__`, and is not something we need to support; although, we should still not crash if there is an invalid value in the `__notes__`. 

I would suggest we just filter out any non-string values from the `__notes__`.

**Comment by szokeasaurusrex** [comment]
Even though the PR says it can be edited by maintainers, I am getting a 403 when I try to push to this PR branch. So, I am closing this PR in favor of #3620

**Pull request closed without merging**